### PR TITLE
Better d-pad align

### PR DESCRIPTION
**Pull request by billwpierce**
Added all 8 angles to align at 45 degrees to better align with boiler

### CODE REVIEW

**Comment by billwpierce** [comment]
I made it use all 8 directions now

File: src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java
Comment (CaseyManning) [inline src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java]: This code seems rather ugly to me. Why don't we have an array of up, left, down, right, etc., and then iterate through them, because it seems like you are doing a similar thing to all of them. You could have a hashmap from position to degrees (eg. [up, 180], [right,90], and so on), and then it would all look much cleaner.

**Review by CaseyManning — CHANGES_REQUESTED** [review]
While this might work, I feel like the code could be cleaner.

File: src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java
Comment (billwpierce) [reply src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java]: @CaseyManning being that I don't know to do what you
're asking and I have stuff to do on r2, I'm going to hold off on doing that for now. It feels like it might not be the best use of time.

Comment (CaseyManning) [reply src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java]: @billwpierce  I agree with you on that. Given the limited amount of time, we can return to this after build season is over if we want to clean it up. For now, if it works, it can be good enough.

**Review by CaseyManning — APPROVED** [review]

File: src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java
Comment (billwpierce) [reply src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java]: yeah. I don't have much to do so I'll start looking at it. That said, we should make sure we have this version of the code readily available so I can test to see if my hashmap works.

Comment (ajnadel) [reply src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java]: @billwpierce I agree with @CaseyManning that this is a bit tedious in terms of readability. What I would suggest is that you make a factory function -- something private in the class that you can call with just an angle measure and it returns a new ChassisTurnAbsolute() with the rest of the parameters taken care of.

**Comment by ajnadel** [comment]
@billwpierce: merge conflicts. When you're updating `standard`, note that the order of arguments in the constructor of `SetEnableableModifier` has changed as of RoboticsTeam4904/standard#171.

**Comment by ajnadel** [comment]
Also, I like the idea of this overall. I'm concerned that it's really difficult to cleanly press a non-cardinal direction on the d-pad (for example, down-right). This can be tested, but I have a feeling that letting go of down-right on the d-pad will cause a twitch towards either down or right.

**Comment by billwpierce** [comment]
@ajnadel we tested it ad it seems pretty reliable. I will fix merge conflicts.

**Comment by billwpierce** [comment]
@ajnadel merge conflicts don't still exist. Once we test we will merge.

File: src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java
Comment (carturn) [reply src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java]: I like @ajnadel idea for this. It is really annoying to have 4 long lines of code that change only one parameter. I am inclined to think that a HashMap will just move the problem somewhere else though (as it will have to be constructed), while the function should reduce it to one or two lines.

**Comment by billwpierce** [comment]
@ajnadel I believe I did what you asked with the function

File: src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java
Comment (ajnadel) [inline src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java]: Nice job abstracting this – just make the capitalization consistent: `bindDPadCommand`.

File: src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java
Comment (ajnadel) [inline src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java]: This was an existing error, but `RobotMap.Component.chassisDriveMC` refers to distance PID using encoders. This should be `RobotMap.Component.chassisTurnMC`, which utilizes the NavX as a sensor.

I don't believe it will work at all as is, unless at some point `chassisDriveMC` was incorrectly set up to use the NavX.

**Review by ajnadel — CHANGES_REQUESTED** [review]

**Comment by c0d3rman** [comment]
The driver does not seem to be using the dpad at all.

**Pull request closed without merging**
